doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,530 @@
+   </style>
+
+.. rst-class:: tb
+
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+| **Simulation technique** | **Memory**                                | **Time**                                       | **Differentiation support**               | **Supports shots/analytic mode** |
++==========================+===========================================+================================================+===========================================+==================================+
+| Deferred measurements    | :rd:`\ ` :math:`\mathcal{O}(2^{n_{MCM}})` | :rd:`\ ` :math:`\mathcal{O}(2^{n_{MCM}})`      | :gr:`\ ` yes                              | :gr:`\ ` yes / yes               |
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+| Dynamic one-shot         | :gr:`\ ` :math:`\mathcal{O}(1)`           | :rd:`\ ` :math:`\mathcal{O}(n_{shots})`        | :or:`\ ` finite differences\ :math:`{}^1` | :or:`\ ` yes / no                |
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+| Tree-traversal           | :or:`\ ` :math:`\mathcal{O}(n_{MCM}+1)`   | :or:`\ ` :math:`\mathcal{O}(\leq 2^{n_{MCM}})` | :or:`\ ` finite differences\ :math:`{}^1` | :or:`\ ` yes / no                |
++--------------------------+-------------------------------------------+------------------------------------------------+-------------------------------------------+----------------------------------+
+
+:math:`{}^1` In principle, parameter-shift differentiation is supported as long as no
+postselection is used. Parameters within conditionally applied operations will
+fall back to finite differences, so that a proper value for ``h`` should be provided, see
+:func:`~.pennylane.gradients.finite_diff`.
+
+The strengths and weaknesses of the simulation techniques differ strongly and the best
+technique will depend on details of the simulation workflow. As a rule of thumb:
+
+- dynamic one-shot sampling excels in the many-measurements-few-shots regime,
+
+- the tree-traversal technique can handle large-scale simulations with many shots
+  and measurements, and
+
+- deferred measurements are the generalist solution that enables mid-circuit measurement
+  support under (almost) all circumstances, but at large memory cost.
+
+By default, ``QNode``\ s use deferred measurements and dynamic one-shot sampling (if supported)
+when executed without and with shots, respectively.
+
+.. _deferred_measurements:
+
+Deferred measurements
+*********************
+
```

Comment:
@dwierichs at this point down, the tone of the page changes somewhat -- it not longer reads so much as a quick reference starting guide, but becomes much more information dense and a bit difficult to read 🤔 

Is it possible to reduce the text in these sections, to make it short and concise?

Reply:
I'll reduce it some! I think it is fair that the tone changes a bit, because the topic itself becomes more technical (not about what PL can do, but how it does it), but I agree that it should read as one cohesive page!

Reply:
@vincentmr I cut down the tree-traversal section significantly, to keep it more in line with a quickstart. Let me know what you think, and whether any of the removed parts should go somewhere else in the docs :)

Reply:
Concise and captures the gist of it, reads well. Thanks @dwierichs .